skrub/_table_vectorizer.py

```diff
@@ -490,8 +490,10 @@ def _auto_cast(self, X, reset=True):
                 # Cast pandas dtypes to numpy dtypes for earlier versions of sklearn.
                 # Categorical dtypes don't need to be casted.
                 # Note that 'is_category_dtype' is deprecated.
-                if is_extension_array_dtype(X[col]) and not isinstance(
-                    X[col].dtype, CategoricalDtype
+                if (
+                    is_extension_array_dtype(X[col])
+                    and not isinstance(X[col].dtype, CategoricalDtype)
+                    and not is_datetime64_any_dtype(X[col])
```

Comment:
there may be some other problematic types we miss in this list. but at this stage it would probably be better to have it crash and users report it than ignore the error

Reply:
I'd rather have errors and fix them than ignore them all.